app/helpers/elasticsearch_helper.rb

```diff
@@ -32,4 +34,10 @@ def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys)
     end
     matching_taxa.values
   end
+
+  private
+
+  def sanitize(prefix)
+    prefix.gsub(/[^a-zA-Z0-9 _|]/, "") if prefix
```

Comment:
You should add at least `.` to the mix.
Even then, I think the best solution is to escape the special characters: http://lucene.apache.org/core/3_0_3/queryparsersyntax.html#Escaping%20Special%20Characters

Reply:
👍 Ok modified this to add backslash in front of special characters. Tested that it works for ones like idseq-bench-4|

Side note, posts like this one point out that query_string is the most raw field http://www.recursion.org/query-parser/#user_input_may_contain_special_characters